examples/test/library_tour/html_test.dart

```diff
@@ -0,0 +1,212 @@
+      if (req.status == 200) {
+        // Successful URL access...
+      }
+      // ...
+      // #enddocregion request
+      expect(req.status, 200);
+      // #docregion request
+    }
+    // #enddocregion request
+
+    await main();
+  });
+
+  test('POST', () async {
+    final url = 'http://httpbin.org/post';
+    // #docregion POST
+    String encodeMap(Map data) => data.keys
+        .map((k) =>
+            '${Uri.encodeComponent(k)}=${Uri.encodeComponent(data[k])}')
+        .join('&');
+
+    Future main() async {
+      var data = {'dart': 'fun', 'angular': 'productive'};
+
+      var request = new HttpRequest();
+      request
+        ..open('POST', url)
+        ..setRequestHeader(
+          'Content-type',
+          'application/x-www-form-urlencoded',
+        )
+        ..send(encodeMap(data));
+
+      await request.onLoadEnd.first;
+
+      if (request.status == 200) {
+        // Successful URL access...
+      }
+      // ...
+      // #enddocregion POST
+      print('Request status: ${request.status}');
+      // #docregion POST
+    }
+    // #enddocregion POST
+
+    expect(main, prints('Request status: 200\n'));
+  });
+
+  test('WebSockets', () async {
+    final logger = new Logger();
+    final print = logger.print; // shadow global print
+    final wsStream = new StreamController<String>();
+
+    // #docregion WebSocket
+    var ws = new WebSocket('ws://echo.websocket.org');
+    // #enddocregion WebSocket
+
+    // #docregion initWebSocket
+    void initWebSocket([int retrySeconds = 1]) {
+      var reconnectScheduled = false;
+
+      print("Connecting to websocket");
+
+      void scheduleReconnect() {
+        if (!reconnectScheduled) {
+          new Timer(new Duration(seconds: retrySeconds),
+              () => initWebSocket(retrySeconds * 2));
+        }
+        reconnectScheduled = true;
+      }
+
+      ws.onOpen.listen((e) {
+        print('Connected');
+        // #docregion send
+        ws.send('Hello from Dart!');
+        // #enddocregion send
+      });
+
+      ws.onClose.listen((e) {
+        print('Websocket closed, retrying in ' +
+            '$retrySeconds seconds');
+        scheduleReconnect();
+      });
+
+      ws.onError.listen((e) {
+        print("Error connecting to ws");
+        scheduleReconnect();
+      });
+
+      // #docregion onMessage
+      ws.onMessage.listen((MessageEvent e) {
+        print('Received message: ${e.data}');
+        // #enddocregion initWebSocket, onMessage
+        wsStream.add(('Received message'));
+        // #docregion initWebSocket, onMessage
+      });
+      // #enddocregion onMessage
+    }
+    // #enddocregion initWebSocket
+
+    final t = wsStream.stream.timeout(
+      new Duration(seconds: 1),
+      onTimeout: (s) => s.add('Timeout!'),
+    );
+
+    try {
+      initWebSocket();
+      expect(await t.first, contains('Received message'));
+    } finally {
+      await wsStream.close();
+    }
+  });
+
+  group('xxxx', () {
```

Comment:
Is this supposed to be here? Should the name be a bit too descriptive? Just checking.

Reply:
Oops, no; the 'xxxx' entries are just templates I use when copy-pasting new code excerpts. They'll be removed in the next round (and by the time we're done with the Library Tour they should be gone for good).